src/Serializer/Filter/GroupFilter.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Serializer\Filter;
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Group filter.
+ *
+ * @author Baptiste Meyer <baptiste@les-tilleuls.coop>
+ */
+final class GroupFilter implements FilterInterface
+{
+    private $overrideDefaultGroups;
+    private $parameterName;
+
+    public function __construct(string $parameterName = 'groups', bool $overrideDefaultGroups = false)
+    {
+        $this->overrideDefaultGroups = $overrideDefaultGroups;
+        $this->parameterName = $parameterName;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function apply(Request $request, bool $normalization, array $attributes, array &$context)
+    {
+        if (!is_array($groups = $request->query->get($this->parameterName))) {
+            return;
+        }
+
+        if (!$this->overrideDefaultGroups && isset($context['groups']) && is_array($context['groups'])) {
```

Comment:
About `is_array($context['groups'])`, I'm not really sure how we should handle the case where `$context['groups']` exists and is not an array:
* do not apply the filter when `$overrideDefaultGroups` is true or false
* apply the filter when `$overrideDefaultGroups` is true
* apply the filter and overwrite the value of `$context['groups']` in all cases (current behavior)

WDYT?